src/styles/main.scss

```diff
@@ -1,3 +1,56 @@
+@import "util/variables";
+@import "util/functions";
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  gap: $gap * 1px;
+  flex-wrap: wrap;
+  width: getSize($base-size, $main-indent, $gap, $columns);
+  padding: $main-indent;
+  margin: 0 auto;
+
+  @each $day-name, $week-num in $start-day {
+    &--start-day-#{$day-name} {
+      & :first-child {
+        margin-left: getFirstDayIndent($week-num, $base-size, $gap);
+      }
+    }
+  }
+
+  @each $last-days in $month-length {
+    &--month-length-#{$last-days} {
+      @for $day from 1 through $last-days {
+        & :nth-child(#{$day}) {
+          background: #eee;
+          border: 1px solid #000;
+          width: $base-size;
+          height: $base-size;
+          box-sizing: border-box;
+
+          &::before {
+            content: "#{$day}";
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            height: 100%;
+            font-family: Arial, Helvetica, sans-serif;
+            font-size: 30px;
+          }
+        }
```

Comment:
You don't need to specify each `:nth-child` as they are the same, you can add these properties to the `&__day`.
One thing that needs to be set through a loop is `content: "#{$day}";`